src/mo_frontend/test_recovery.ml

```diff
@@ -0,0 +1,457 @@
+
+let parse_from_lexbuf lexbuf : Mo_def.Syntax.prog Diag.result =
+  let open Mo_frontend in
+  let open Diag.Syntax in
+  let name = "test.mo" in
+  let lexer_mode = Lexer.mode in
+  let lexer, triv_table = Lexer.tokenizer lexer_mode lexbuf in
+  let () = Parser_lib.triv_table := triv_table in
+  let start =  Parser.Incremental.parse_prog lexbuf.Lexing.lex_start_p in
+  let error_details = !Mo_config.Flags.error_detail in
+  let* mk_syntax = Mo_frontend.Parsing.parse ~recovery:true lexer_mode error_details start lexer lexbuf
```

Comment:
These test might be a bit unmaintable and break every time the grammar changes. 
Maybe it would be better to dial down the `error_detail` so that only the unexpected token is reported and you don't get all the extra info in the expected test output.

Just a suggestion, but I think this might be more maintainable (unless you are actually testing the error output rather than the recovered AST).

Reply:
I was using `dune runtest --auto-promote`. Does it solve the maintenance problem?

Reply:
Oh, I doubt anyone is aware of that. Maybe just add a comment to use that at the top of the file?

Reply:
done